packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts

```diff
@@ -216,23 +230,71 @@ export class Pipeline extends cdk.Construct implements events.IEventRuleTarget {
      *       onChildAdded type hook.
      * @override
      */
-    protected addChild(child: cdk.Construct, name: string) {
-        super.addChild(child, name);
+    protected addChild(child: cdk.Construct, name: string, props?: any) {
```

Comment:
I don’t think this change is needed and we are extremely sensitive to extensions of the surface area of Construct. 

I see why you need to propagate props to `addChild` but that’s not a robust way to do this. What promises you that everyone will pass in their props to `Construct`?

I think the way to address this is to render the stages array only during synthesis (using a `Token`). Then, you wouldn’t even need to override `addChild` at all. When you synthesize, you will iterate over your children and devise the list of stages according to their placement.

Reply:
Changes in Construct are not needed anymore after #568.